client/src/newsletter/PermissionCreator.tsx

```diff
@@ -0,0 +1,256 @@
+  type: 'SERVICE' | 'SALES' | 'PRODUCT' | 'NEWSLETTER' | 'default';
+  hidden: boolean;
+  description: string;
+  consentText: string;
+};
+
+type AlertsState = { status: 'error' | 'success' | null; msg: string };
+
+type InitState = {
+  values: ValuesState;
+  alert: AlertsState;
+};
+
+const initState = {
+  values: {
+    name: '',
+    frequency: '',
+    brandId: 'default',
+    type: 'default',
+    hidden: false,
+    description: '',
+    consentText: '',
+  },
+  alert: {
+    status: null,
+    msg: '',
+  },
+} as InitState;
+
+export const PermissionCreator = ({
+  handleCancel,
+}: PermissionCreatorProps): JSX.Element => {
+  const [values, setValues] = React.useState<ValuesState>(initState.values);
+  const [alert, setAlert] = React.useState<AlertsState>(initState.alert);
+  const [isProcessing, setProcessing] = React.useState<boolean>(false);
+  const newsletterSdk = useNewsletterSdk();
+
+  React.useEffect(() => {
+    let timeoutId: NodeJS.Timeout;
+    if (alert.status === 'success') {
+      timeoutId = setTimeout(() => {
+        handleCancel();
+      }, 2000);
+    }
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [alert.status]);
+
+  const getFormValidation = () => {
+    let validate = true;
+    const keys = Object.keys(values).filter(
+      (key) => key !== 'frequency' && key !== 'hidden',
+    );
+
+    keys.forEach((key) => {
+      if (validate) {
+        validate = values[key] !== '' && values[key] !== 'default';
+      }
+    });
+
+    if (values.type === 'NEWSLETTER') {
+      validate = !!values.frequency;
+    }
+
+    return validate;
+  };
+
+  const createPermission = async () => {
+    const data = {
+      ...values,
+      brandId: parseInt(values.brandId.toString(), 10),
+      enabled: true,
+      sortId: 0,
+    } as CreatePermissionDto;
+    try {
+      await newsletterSdk.permissions.admin.create(data);
+      setAlert({ status: 'success', msg: 'Created permission' });
+    } catch (e) {
+      setAlert({ status: 'error', msg: e.message });
+    } finally {
+      setProcessing(false);
+    }
+  };
+
+  const create = (event: React.FormEvent) => {
+    event.preventDefault();
+    setProcessing(true);
+
+    if (getFormValidation()) {
+      createPermission();
+    } else {
+      setAlert({ status: 'error', msg: 'Fill form correctly' });
+      setProcessing(false);
+    }
+  };
+
+  const handleOnChange = (event: React.ChangeEvent<HandleOnChange>) => {
+    const { value, name, type } = event.currentTarget as HandleOnChange;
+    const { checked } = event.currentTarget as HTMLInputElement;
+
+    setValues((prevState) => ({
+      ...prevState,
+      [name]:
+        type.toLowerCase() === 'checkbox' && name == 'hidden' ? checked : value,
+      frequency:
+        name === 'type' && value !== 'newsletter'
+          ? ''
+          : name === 'frequency'
+          ? value
+          : prevState.frequency,
+    }));
+    setAlert(initState.alert);
+  };
```

Comment:
make it more readable by extracting long conditions to variables

```suggestion
 const handleFormElementChange = (event: React.ChangeEvent<FormElement>) => {
  const { value, name, type } = event.currentTarget as FormElement;
  const { checked } = event.currentTarget as HTMLInputElement;

  if (name === 'type' && value === 'newsletter') {
    return setFormValues((prevState: FormValues) => ({
      ...prevState,
      type: 'NEWSLETTER',
      frequency: '',
    }));
  }

  const newValue = type.toLowerCase() === 'checkbox' ? checked : value;

  setFormValues((prevState: FormValues) => ({
    ...prevState,
    [name]: newValue,
  }));
};
```